benches/containerd-shim-benchmarks/benches/webassembly-benchmarks.rs

```diff
@@ -0,0 +1,362 @@
+    suite.
+
+    Because of this, only select the benchmarks which run in under 5 seconds on
+    a desktop computer using WasmEdge. Note that this selection is pretty
+    arbitrary and we can add or remove benchmarks easily, also from other
+    sources. This is the selection algorithm:
+
+    $ for file in *; do if timeout 5s wasmedge "${file}" > /dev/null; then echo "$file"; fi; done
+    aead_chacha20poly13052.wasm
+    aead_chacha20poly1305.wasm
+    aead_xchacha20poly1305.wasm
+    auth2.wasm
+    auth3.wasm
+    auth6.wasm
+    auth.wasm
+    box_seed.wasm
+    generichash2.wasm
+    generichash3.wasm
+    hash3.wasm
+    hash.wasm
+    kdf.wasm
+    keygen.wasm
+    onetimeauth2.wasm
+    onetimeauth.wasm
+    scalarmult2.wasm
+    scalarmult5.wasm
+    scalarmult6.wasm
+    secretbox2.wasm
+    secretbox_easy.wasm
+    secretbox.wasm
+    secretstream_xchacha20poly1305.wasm
+    shorthash.wasm
+    siphashx24.wasm
+    stream3.wasm
+    stream4.wasm
+
+    Criterion is set to run each benchmark ten times, ten being the minimum
+    number of iterations that Criterion accepts. If we need more statistically
+    meaningful results we can increase the number of iterations (with the cost
+    of a longer benchmarking time). Running the whole suite on a desktop
+    computer takes now a bit over 10 minutes.
+*/
+
+#[derive(Serialize, Deserialize)]
+struct Options {
+    root: Option<PathBuf>,
+}
+
+fn get_external_benchmark_module(name: String) -> Result<Vec<u8>, Error> {
+    let manifest_dir = env!("CARGO_MANIFEST_DIR");
+    let target = Path::new(manifest_dir)
+        .join("../../benches/webassembly-benchmarks/2022-12/wasm")
+        .join(name.clone());
+    std::fs::read(target).map_err(|e| {
+        Error::Others(format!(
+            "failed to read requested Wasm module ({}): {}.",
+            name, e
+        ))
+    })
+}
+
+fn run_wasmtime_test_with_spec(
+    dir: &TempDir,
+    spec: &Spec,
+    wasmbytes: &[u8],
+) -> Result<(u32, DateTime<Utc>), Error> {
+    create_dir(dir.path().join("rootfs"))?;
+
+    let wasm_path = dir.path().join("rootfs/file.wasm");
+    let mut f = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .mode(0o755)
+        .open(wasm_path)?;
+    f.write_all(wasmbytes)?;
+
+    let stdout = File::create(dir.path().join("stdout"))?;
+    drop(stdout);
+
+    spec.save(dir.path().join("config.json"))?;
+
+    let mut cfg = InstanceConfig::new(WasmtimeWasi::new_engine()?, "test_namespace".into());
+    let cfg = cfg
+        .set_bundle(dir.path().to_str().unwrap().to_string())
+        .set_stdout(dir.path().join("stdout").to_str().unwrap().to_string());
+
+    let wasi = Arc::new(WasmtimeWasi::new("test".to_string(), Some(cfg)));
+
+    wasi.start()?;
+
+    let w = wasi.clone();
+    let (tx, rx) = channel();
+    let waiter = Wait::new(tx);
+    w.wait(&waiter).unwrap();
+
+    let res = match rx.recv_timeout(Duration::from_secs(600)) {
```

Comment:
10 minutes of timeout seems too long for me...

Reply:
That's right. I reduced the timeout to one minute. That should be plenty too...